src/ngraph/runtime/gpu/CMakeLists.txt

```diff
@@ -72,8 +72,32 @@ if (NGRAPH_GPU_ENABLE)
 	    -gencode=arch=compute_70,code=compute_70)
     endif()
 
-    set_source_files_properties( ${CUDA_SRC} PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)
-    cuda_compile(CUDA_OBJ ${CUDA_SRC} STATIC)
+    set (DO_CUDA_COMPILE FALSE)
+    if (CUDA9_FOUND)
+        set (DO_CUDA_COMPILE TRUE)
+        if (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"))
+            # CUDA 9 supports up to gcc 6.x
+            if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+                set (DO_CUDA_COMPILE TRUE)
+            endif()
+        # CUDA 9 supports up to clang 3.9
+        elseif (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.0)
+            set (DO_CUDA_COMPILE TRUE)
+        endif()
+    elseif (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"))
+        # Current release of CUDA 8 supports up to gcc 5.4
+        if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.5)
+            set (DO_CUDA_COMPILE TRUE)
+        endif()
+    # CUDA 8 supports up to clang 3.8
+    elseif(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.9)
```

Comment:
Should check for clang, not just assume it is clang. There are other compilers that we currently don't use like windows and icc.

Reply:
Ok - will rework. Thought maybe I was safe given this warning from another CMakeLists.txt:
message(WARNING "You are using an unsupported compiler. Compilation has only been tested with Clang (${CLANG_MIN_VERSION} and up), Apple Clang (${APPLE_CLANG_MIN_VERSION} and up), and GCC (${GCC_MIN_VERSION} and up).")